src/modules/MagneticFieldReader/MagneticFieldReaderModule.cpp

```diff
@@ -62,5 +62,87 @@ void MagneticFieldReaderModule::initialize() {
                        << Units::display(detector->getMagneticField(detector->getLocalPosition(position)), {"T", "mT"});
         }
         LOG(INFO) << "Set constant magnetic field: " << Units::display(b_field, {"T", "mT"});
+    } else if(field_model == MagneticField::MESH) {
+        type = MagneticFieldType::CUSTOM;
+
+        auto fallback_field = config_.get<ROOT::Math::XYZVector>("magnetic_field_fallback", ROOT::Math::XYZVector());
+        auto field_data = read_field();
+
+        auto dims = field_data.getDimensionality(); // Vector dimensionality of the field (should be 3 for B-field)
+        if(dims != 3) {
+            throw std::invalid_argument("B-field vectors were not three dimensional.");
```

Comment:
I would prefer something along the lines of

```suggestion
            throw InvalidValueError(config_, "file_name", "B-field vectors were not three dimensional");
```

Because this gives a meaningful error message pointing the user to the correct place directly.

Reply:
...moreover I would throw this directly in `read_file()` I think, because that's the earliest place you can detect it. I'm also not sure if this can happen inf you read `FieldQuantity::VECTOR`, doesn't the parsing code then mandate 3 dimensions already?

Reply:
You're right, there are already checks (and error handling) for dimensionality etc. in the file parsing code!